kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -56,6 +56,39 @@ class RobotState;
 namespace kinematics
 {
 
+/*
+ * @enum KinematicSearches
+ * @brief Flags for choosing the type discretization applied on the redundancy during a multiple ik solution search
+ */
+namespace KinematicSearches
+{
+  enum KinematicSearch
+  {
+    ONE = 1,
+    ALL_DISCRETIZED ,
+    SOME_DISCRETIZED,
+    RANDOM_SAMPLE,
+    ALL
+  };
+}
+typedef KinematicSearches::KinematicSearch KinematicSearch;
+
+/*
+ * @enum KinematicErrors
+ * @brief Kinematic error codes that occur in a multiple ik solution search
+ */
+namespace KinematicErrors
+{
+  enum KinematicError
+  {
+    OK = 1,
+    UNSUPORTED_SEARCH_REQUESTED,
+    DISCRETIZATION_NOT_INITIALIZED,
+    NO_SOLUTION
```

Comment:
I think adding codes to the existing error codes is possible.  If we group all IK failures, whether they are caused by disretization or not, into a single error code, then the only possible error handling is too generic to be of much use.  From a user/developer standpoint, it would help to know if the failure is due to discretization or if the pose is in fact unreachable.